examples/rosettacode/24_game.nit

```diff
@@ -0,0 +1,157 @@
+#!/usr/bin/env nit
+#
+# This file is part of NIT ( http://www.nitlanguage.org ).
+# This program is public domain
+
+# Task: 24 game
+# SEE: <http://rosettacode.org/wiki/24_game>
+
+# Export `operands` and `numbers` from string `operation`
+fun exportation(operation: String, numbers: Array[Int], operands: Array[Char]) do
+	var previous_char: nullable Char = null
+	var number: nullable Int = null
+	var negative = false
+
+	for i in operation.length.times do
+		var current_char = operation[i]
+		var current_int = current_char.to_i
+
+		if (previous_char == null or "-+/*".has(previous_char)) and current_char == '-' then
+			negative = true
+			continue
+		end
+
+		if current_int >= 0 and current_int <= 9 then
+			if number == null then
+				number = current_int
+			else
+				number = number * 10 + current_int
+			end
+		end
+
+		if negative and ("-+/*".has(current_char) or i == operation.length - 1) then
+			number = number - number * 2
+			negative = false
+		end
+
+		if ("-+/*".has(current_char) or i == operation.length - 1) and number != null then
+			numbers.add(number)
+			number = null
+		end
+
+		if not negative and "-+/*".has(current_char) then
+			operands.add(current_char)
+		end
+		previous_char = current_char
+	end
+end
+
+# Create `random_numbers` from 0 to 9
+fun random(random_numbers: Array[Int]) do
+	for i in [0..4[ do
+		var number = 0
+		while number == 0 do number = 10.rand
+		random_numbers.add(number)
+	end
+end
+
+# Make calculus with `numbers` and `operands` and add the operation result into `random_numbers` 
+fun calculus(random_numbers, numbers: Array[Int], operands: Array[Char]) do
+	var number = 0
+	var temp_numbers = numbers.clone
+
+	while temp_numbers.length > 1 do
+		var operand = operands.first
+
+		if operand == '+' then number = temp_numbers.first + temp_numbers[1]
+		if operand == '-' then number = temp_numbers.first - temp_numbers[1]
+		if operand == '*' then number = temp_numbers.first * temp_numbers[1] 
+		if operand == '/' then number = temp_numbers.first / temp_numbers[1]
+		
+		operands.remove_at(0)
+		temp_numbers.insert(number, 0)
+		
+		for i in [0..1] do temp_numbers.remove_at(1)
```

Comment:
This is hard to read, the following may be a bit better:
~~~
var operand = operands.shift
var a = temp_numbers.shift
var b = temp_numbers.shift

if operand == '+' then number = a + b
if operand == '-' then number = a - b
if operand == '*' then number = a * b
if operand == '/' then number = a / b

temp_numbers.unshift number
~~~